packages/schemas/src/foundations/jsonb-types.ts

```diff
@@ -31,3 +31,51 @@ export const userLogPayloadGuard = z.object({
 });
 
 export type UserLogPayload = z.infer<typeof userLogPayloadGuard>;
+
+export const signingKeyPayload = z.object({
+  keyId: z.string(),
+});
+
+export const signingKeyManagementPayloadGuard = z.object({
+  signing_algorithm: z.enum(['RS256']),
+  signing_keys: z.object({
+    previously_used_keys: signingKeyPayload.array(),
+    currently_used_key: signingKeyPayload,
+    next_key_in_queue: signingKeyPayload,
+  }),
+});
+
+export type SigningKeyManagementPayload = z.infer<typeof signingKeyManagementPayloadGuard>;
+
+export const apiAuthorizationSettingsPayloadGuard = z.object({
+  default_audience: z.string().url({ message: 'Invalid URL' }), // Default oidc audience (usually client_id).
+  default_directory: z.enum(['logto', 'email', 'sms', 'waad', 'adfs']), // Name of the connection to be use for Password Grant exchanges.
+});
+
+export type ApiAuthorizationSettingsPayload = z.infer<typeof apiAuthorizationSettingsPayloadGuard>;
+
+export const loginSessionManagementPayload = z.object({
+  session_cookie_mode: z.enum(['persistent', 'non-persistent']),
+  /**
+   * Persistent Session: Allows the user to retain their session cookie when re-opening the browser on the same device.
+   * Non-Persistent Session: Invalidates the session cookie when the browser is closed.
+   */
+  inactivity_timeout: z
+    .number()
+    .int()
+    .positive()
+    .lte(24 * 60 * 3), // In minutes, max 3 days (can be configurable in next version).
+  require_login_after: z
+    .number()
+    .int()
+    .positive()
+    .lte(24 * 60 * 30), // In minutes, max 30 days (can be configurable in next version).
+});
+
+export const loginAndLogoutPayloadGuard = z.object({
+  login_url: z.string().url({ message: 'Invalid URL' }), // This URI needs to point to a route in your application that should redirect to your tenant's `/authorize` endpoint.
+  logout_urls: z.string().url({ message: 'Invalid URL' }).array(), // After users log out, users can be redirected to a specific URL. This configuration can be set here or in application settings.
+  login_session_management: loginSessionManagementPayload,
+});
```

Comment:
looks like it's a per-app config?

Reply:
`logout_urls` is truly per-app config. we do not have authenticate timeout config for apps now, it can be customized both globally and locally (if not specified for the specific app then global setting applied). Shall we add `authentication_timeout` config for each app or just assume that each Logto user owns exact one app?

Reply:
- if it's a per-app config, let's remove it for now
- i think the assumption is not valid even for v1